Fix for the valid_range attribute definition in GNSSRO converters

fabiolrdiniz opened the pull request:
## Description
This is a PR proposing a fix for the `valid_range` attribute definition being written by GNSSRO converters.

P.S.: this PR is also addressing the comments from @jamiebresch after #766 has been merged into develop (see https://github.com/JCSDA-internal/ioda-converters/pull/766#discussion_r798867742 and https://github.com/JCSDA-internal/ioda-converters/pull/766#discussion_r798868919).

### Issue(s) addressed
- fixes #779

## Acceptance Criteria (Definition of Done)
Redefines the `valid_range` attribute.

## Dependencies
None

## Impact
None

jamiebresch commented:
There could warnings for line 363 and line 392 `Warning: Missing '&' in continued character constant at (1) [-Wampersand]`.
Either add `&` before `e.g.`, or break the previous lines in the beginning of the whole string.

jamiebresch commented on src/gnssro/gnssro_bufr2ioda.f90:
```diff
@@ -301,7 +301,7 @@ program gnssro_bufr2ioda2
        gnssro_data%bend_ang(ndata)     = bend
        gnssro_data%bndoe_gsi(ndata)    = bend_error
-       gnssro_data%impact_para(ndata)  = impact! 
+       gnssro_data%impact_para(ndata)  = impact!
```
remove `!` as well?

fabiolrdiniz replied:
Sure! The space has been removed automatically by my text editor. I just removed it by hand.

jamiebresch commented:
src/ncar-bufr2nc-fortran/gnssro_mod.f90 is out-of-sync with gnssro_bufr2ioda.f90. A few more changes are needed to bring it up-to-date. Is there a need to maintain src/ncar-bufr2nc-fortran?

hailingz commented:
Thank you very much Fabio for addressing those issues!
do you know why the ctest failed?

hailingz commented on src/gnssro/gnssro_bufr2ioda.f90:
```diff
@@ -381,7 +381,9 @@ program gnssro_bufr2ioda2
 call check( nf90_def_var(grpid_metadata, "ascending_flag",  NF90_INT, nlocs_dimid, varid_asce))
 call check( nf90_put_att(grpid_metadata, varid_asce, "longname", "the original occultation ascending/descending flag" ))
-call check( nf90_put_att(grpid_metadata, varid_asce, "valid_range", "0/descending or 1/ascending" ))
+call check( nf90_put_att(grpid_metadata, varid_asce, "valid_range", int((/ 0, 1 /))) )
+call check( nf90_put_att(grpid_metadata, varid_asce, "flag_values", int((/ 0, 1 /))) )
+call check( nf90_put_att(grpid_metadata, varid_asce, "flag_meanings", "descending ascending") )
```
This is very nice!

fabiolrdiniz replied:
> src/ncar-bufr2nc-fortran/gnssro_mod.f90 is out-of-sync with gnssro_bufr2ioda.f90. A few more changes are needed to bring it up-to-date. Is there a need to maintain src/ncar-bufr2nc-fortran?

Thanks again, @jamiebresch. I just fixed the other warning for `qfro` there. I'll leave the question about the maintenance of `src/ncar-bufr2nc-fortran` for the others because I'm not the right person to say about it.

fabiolrdiniz commented:
> Thank you very much Fabio for addressing those issues! do you know why the ctest failed?

Yes, I know. I forgot to update the reference files after adding the flag values & meanings. Sorry reviewers! It should pass the CI tests.

fabiolrdiniz commented:
> src/ncar-bufr2nc-fortran/gnssro_mod.f90 is out-of-sync with gnssro_bufr2ioda.f90. A few more changes are needed to bring it up-to-date. Is there a need to maintain src/ncar-bufr2nc-fortran?

Thanks again, @jamiebresch. I just fixed the other warning for `qfro` there. I'll leave the question about the maintenance of `src/ncar-bufr2nc-fortran` for the others because I'm not the right person to say about it.

jamiebresch commented:
Since you are touching src/ncar-bufr2nc-fortran/gnssro_mod.f90, you can grab an updated version from https://github.com/jamiebresch/obs2ioda/blob/main/obs2ioda-v1/src/gnssro_mod.f90 if you want, but you will need to redo your valid_range changes.
I would propose to remove src/ncar-bufr2nc-fortran/gnssro_mod.f90 if it is not used.

fabiolrdiniz commented:
@jamiebresch, I just pushed a version of the code following your suggestions for updating `src/ncar-bufr2nc-fortran/gnssro_mod.f90` and fixing the ampersand issue above.

hailingz reviewed (approved):
The GNSS RO converters look good to me!

jamiebresch commented:
Thanks @fabiolrdiniz for addressing my concerns and being so thorough.

srherbener reviewed (approved):
It looks like the `ncar-bufr2nc-fortran/gnssro_mod.f90` is still outputting ioda v1 format. I think dealing with upgrading this converter to ioda v2 can be done in a subsequent PR. Ie, keep this PR focused on the fixes for the valid_range attribute.

What do others think?

srherbener commented:
@jamiebresch are you okay with merging, or do you still have some comments you want addressed? Thanks!

jamiebresch commented:
The fixes look good to me.
